bundles/org.openhab.binding.nobohub/src/main/java/org/openhab/binding/nobohub/internal/discovery/NoboThingDiscoveryService.java

```diff
@@ -0,0 +1,161 @@
+        bridgeHandler.startScan();
+    }
+
+    @Override
+    public synchronized void stopScan() {
+        super.stopScan();
+        removeOlderResults(getTimestampOfLastScan());
+    }
+
+    @Override
+    public void deactivate() {
+        removeOlderResults(new Date().getTime());
+    }
+
+    public void detectZones(Collection<Zone> zones) {
+        ThingUID bridge = bridgeHandler.getThing().getBridgeUID();
+        List<Thing> things = bridgeHandler.getThing().getThings();
+
+        for (Zone zone : zones) {
+            if (bridge != null) {
+                ThingUID discoveredThingId = new ThingUID(THING_TYPE_ZONE, bridge, Integer.toString(zone.getId()));
+
+                boolean addDiscoveredZone = true;
+                for (Thing thing : things) {
+                    if (thing.getUID().equals(discoveredThingId)) {
+                        addDiscoveredZone = false;
+                    }
+                }
+
+                if (addDiscoveredZone) {
+                    String label = zone.getName();
+
+                    Map<String, Object> properties = new HashMap<>(1);
+                    properties.put("id", Integer.toString(zone.getId()));
+                    properties.put("name", zone.getName());
+                    properties.put("vendor", VENDOR);
+
+                    logger.debug("Adding device {} to inbox", discoveredThingId);
+                    DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(discoveredThingId)
+                            .withBridge(bridge).withLabel(label).withProperties(properties)
+                            .withRepresentationProperty("id").build();
+                    thingDiscovered(discoveryResult);
+                }
+            }
+        }
+    }
+
+    public void detectComponents(Collection<Component> components) {
+        ThingUID bridge = bridgeHandler.getThing().getBridgeUID();
+        List<Thing> things = bridgeHandler.getThing().getThings();
+
+        for (Component component : components) {
+            if (bridge != null) {
+                ThingUID discoveredThingId = new ThingUID(THING_TYPE_COMPONENT, bridge,
+                        component.getSerialNumber().toString());
+
+                boolean addDiscoveredComponent = true;
+                for (Thing thing : things) {
+                    if (thing.getUID().equals(discoveredThingId)) {
+                        addDiscoveredComponent = false;
+                    }
+                }
+
+                if (addDiscoveredComponent) {
+                    String label = component.getName();
+
+                    Map<String, Object> properties = new HashMap<>(1);
+                    properties.put("serialNumber", component.getSerialNumber().toString());
+                    properties.put("name", component.getName());
+                    properties.put("vendor", VENDOR);
+                    properties.put("model", component.getSerialNumber().getComponentType());
```

Comment:
Please consider constants for consistency.

Reply:
Fixed